### DISCUSSION TITLE
Allow SUPPLY key to adapt metadata of an NFT in the treasury wallet

### DISCUSSION THREAD

**Discussion by HGraphPunks** [hiero-ledger/hiero-improvement-proposals#660] [Ideas]
This HIP is being created to separate the METADATA key HIP and the SUPPLY key adaptation idea into two smaller scoped HIPs instead of all changes going through one HIP.

**Relevant METADATA key HIP discussion:**
https://github.com/hashgraph/hedera-improvement-proposal/discussions/607

**Problem**
There is no method for an NFT to be updated only when in the treasury wallet, while guaranteeing immutability of the metadata URI after the NFT was distributed. 

**Current Solution**
- Have the user send the NFT to the treasury wallet, burn that serial number, mint a new serial number with the new data using the supply key, send back to the user

**Use cases**
Dynamic NFTs can have a lot of use cases as seen on other chains
- A legal agreement NFT that requires updating after mint to reflect new parameters agreed by both parties but cannot be changed once the NFT is no longer in the treasury wallet.
- A game piece NFT that is updated with game stats when playing the game, but cannot be changed once the player takes back ownership of the NFT. Guaranteeing immutable values for the player when interacting with other services on Hedera

**HIP idea**

- Allow SUPPLY key to update and adapt the metadata of a NFT serial number that is in the treasury wallet.
- The requirement of the NFT needing to be in the treasury wallet would protect against any malicious activity with remote changing of a project already distributed.

**Comment by AdrianKBL** [comment]
Good night,

Excellent proposal @HGraphPunks , I totally agree. This is something **totally necessary in Hedera**, we (KabilaApp) are managing many collections of other Chains and this is one of the first things that we are always asked, about the mutability of NFTs. I think it should be incorporated into Hedera and it would help further adoption of NFTs.

Perhaps what I'm not sure about is if it would be best to use the **Supply_Key** or create a new Key to update the **Metadata**.

**Reply by HGraphPunks** [reply]
Hey @AdriMsM91 !!

So I actually created a separate HIP for a METADATA key that is going forward.

You can read the discussions here:
https://github.com/hashgraph/hedera-improvement-proposal/discussions/607

PR here:
https://github.com/hashgraph/hedera-improvement-proposal/pull/659

A METADATA key is going to be created for mass updates to NFT metadata with no restriction along with a new call on the network to adapt serial numbers metadata. The only issue is that there's no way to guarantee that a METADATA key won't be used to update all NFTs at any time. 

I believe there's a use case that isn't covered, where I want to adapt the metadata, but I need my users to know I can't update it when it's in their wallet / distributed on the network. That's where the SUPPLY key makes sense to me.

SUPPLY key can
- Burn NFTs that are in teh treasury wallet
- Mint NFTs into the treasury wallet

I think it should also be able to
- Edit NFTs metadata URI that are in the treasury wallet

**Reply by Ashe-Oro** [reply]
important distinction, thank you.

**Comment by Ashe-Oro** [comment]
The ability to ensure immutability of NFTs for community members/holders while retaining the ability to update metadata while in the Treasury account seems like a very good feature. It both reduces trust while offering the fundamental functionality of dynamic NFTs without requiring the use of a centralized storage solution (ie AWS).

**Comment by HGraphPunks** [comment]
Bumping this one as well. I think this is a well defined goal and segmented enough that it can go forward since much discussion by the community was already had in the previous hip this was removed from. 

@mgarbs @Ashe-Oro Any objections to pushing this forward to the next stage of HIPs? I could do a PR

**Comment by AdrianKBL** [comment]
Hello, good evening!!

Any news about this one?

I think Hedera Ecosystem needs it!

**Reply by Ashe-Oro** [reply]
it's active and being discussed :)
